plugins/patchmanager.koplugin/main.lua

```diff
@@ -0,0 +1,182 @@
+--[[--
+Plugin for automatic dimming of the frontlight after an idle period.
+
+@module koplugin.PatchManager
+--]]--
+
+local DataStorage = require("datastorage")
+local lfs = require("libs/libkoreader-lfs")
+
+if lfs.attributes(DataStorage:getDataDir() .. "/patches", "mode") ~= "directory" then
+    return { disabled = true }
+end
+
+local Device = require("device")
+local InfoMessage = require("ui/widget/infomessage")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local userPatch = require("userpatch")
+local _ = require("gettext")
+
+local Screen = Device.screen
+
+local PatchManager = WidgetContainer:extend{
+    name = "patchmanager",
+    is_doc_only = false,
+}
+
+function PatchManager:init()
+    self.patch_dir = DataStorage:getDataDir() .. "/patches"
+    self.disable_ext = ".disabled"
+    self.patches = {}
+    self:getAvailablePatches()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function PatchManager:getAvailablePatches()
+    for priority = tonumber(userPatch.early_once), tonumber(userPatch.on_exit) do
+        self.patches[priority] = {}
+        for entry in lfs.dir(self.patch_dir) do
+            local mode = lfs.attributes(self.patch_dir .. "/" .. entry, "mode")
+            if entry and mode == "file" and entry:match("^" .. priority .. "%d*%-") then
+                if entry:find(".lua", 1, true) then
+                    -- exclude any non lua files
+                    table.insert(self.patches[priority], entry)
+                end
+            end
+        end
+    end
+end
+
+function PatchManager:getSubMenu(priority)
+    if priority < tonumber(userPatch.early_once) or priority > tonumber(userPatch.on_exit) then
+        logger.err("PatchManager: BUG, wrong patch_level:", priority)
+    end
+    local sub_menu = {}
+    if #self.patches == 0 then
+        return {}
+    end
+    local function getExecutionStatus(patch_name)
+        print("xxx", patch_name)
+        if userPatch.execution_status[patch_name] then
+            return " (success)"
+        else
+            return " (error)"
```

Comment:
You could probably go with Unicode symbols instead.
https://www.fileformat.info/info/unicode/block/dingbats/list.htm
https://www.fileformat.info/info/unicode/block/miscellaneous_symbols/list.htm

(In my own manichaean patch, I use https://www.fileformat.info/info/unicode/char/2605/index.htm and https://www.fileformat.info/info/unicode/char/2716/index.htm for good and evil - but a checkmark is probably better than my star in this context.)

Reply:
I use the menu checkmark to show if a patch is enabled. 
What about 👍 to show success and 👎 for an error?

Reply:
Depend on how these "hand" icons look in our fonts, but it's also not really our "theme" to show such hands, we're rather on the simple symbols (checkmark, star...) side, so try not to go too wild :) (Also, the weight/density of both hands may be to similar to be quickly distinguished - we want the "fail/error" symbol to strike :)
There are also other kind of checkmark symbols that may look different enough than the one used on the left.